src/column.rs

```diff
@@ -469,65 +469,70 @@ impl HashColumn {
 
 	pub fn write_plan(
 		&self,
-		key: &Key,
-		value: Option<&[u8]>,
+		change: &Change<Key, Vec<u8>>,
 		log: &mut LogWriter,
 	) -> Result<PlanOutcome> {
 		let tables = self.tables.upgradable_read();
 		let reindex = self.reindex.upgradable_read();
-		let existing = Self::search_all_indexes(key, &*tables, &*reindex, log)?;
+		let existing = Self::search_all_indexes(change.key(), &*tables, &*reindex, log)?;
 		if let Some((table, sub_index, existing_address)) = existing {
-			self.write_plan_existing(&*tables, key, value, log, table, sub_index, existing_address)
-		} else if let Some(value) = value {
-			let (r, _, _) = self.write_plan_new(tables, reindex, key, value, log)?;
-			Ok(r)
+			self.write_plan_existing(&*tables, change, log, table, sub_index, existing_address)
 		} else {
-			log::trace!(target: "parity-db", "{}: Deleting missing key {}", tables.index.id, hex(key));
-			if self.collect_stats {
-				self.stats.remove_miss();
+			match change {
+				Change::SetValue(key, value) => {
+					let (r, _, _) = self.write_plan_new(tables, reindex, key, value, log)?;
+					Ok(r)
+				},
+				Change::RemoveValue(key) => {
+					log::trace!(target: "parity-db", "{}: Deleting missing key {}", tables.index.id, hex(key));
+					if self.collect_stats {
+						self.stats.remove_miss();
+					}
+					Ok(PlanOutcome::Skipped)
+				},
+				Change::IncRc(key) => {
```

Comment:
This should be logged in a new stats field. 
There's an important difference between using `SetValue` and `IncRc`. The code might check that the value exists and try to increase the reference with `IncRc`, while some other thread commits a transaction that removes the value. In that case `IncRc` will fail, since the value does not exist anymore, whils `SetValue` would work fine. The user of the `IncRc` have to be 100% sure they don't get the value deleted before the trasaction is commited. Having a stat field might help diagnose such issues.

Reply:
I've seen you remove the statistic from the flush of the memmap (makes sense).
I was wondering if maybe we would just want the stats in their own file without using the WAL (having to add a single field is not such an happy perspective to me), so we can make it more flexible.
Could for example just use something like crates binary-layout over a single memmap (no idea if the crate is good I just remember I wanted to test it).

Reply:
Keeping it with the index seemed more simple. For now you can just add the field to the end of the struct and it should not break compatibility with existing data.